code/modules/mob/living/carbon/internals.dm

```diff
@@ -0,0 +1,68 @@
+/mob/living/carbon/proc/has_breathing_mask()
+	return istype(wear_mask, /obj/item/clothing/mask)
+
+/mob/living/carbon/proc/internals_candidates() //These are checked IN ORDER.
+	return get_all_slots() + held_items
+
+/mob/living/carbon/human/internals_candidates() //Humans have a lot of slots, so let's give priority to some of them
+	var/list/priority = list(s_store, back, belt, l_store, r_store)
+	return priority | get_all_slots() | held_items //| operator ensures there are no duplicates
+
+/mob/living/carbon/proc/get_internals_tank()
+	for(var/obj/item/weapon/tank/T in internals_candidates())
+		//We found a tank!
+		if(istype(T, /obj/item/weapon/tank/jetpack)) //Oh... But it's a jetpack... We'll use it if we have to, but let's see if we find something better first
+			if(!.) //We already had another jetpack
+				. = T
+			continue
+		else //It's the real deal!
+			return T
+
+// Set internals on or off.
+/mob/living/carbon/proc/toggle_internals(var/mob/living/user, var/obj/item/weapon/tank/T)
+	if(internal)
+		internal.add_fingerprint(user)
+		internal = null
+		if(internals) //This is the HUD icon, these variables have WAY too similar names
+			internals.icon_state = "internal0"
+		if(user != src)
+			if(!user.isGoodPickpocket())
```

Comment:
I wasn't insulting the code I just thought this is evil. Taking people's internals?